src/overlays/actors/ovl_En_Nb/z_en_nb.c

```diff
@@ -21,255 +169,1401 @@ const ActorInit En_Nb_InitVars = {
+    Player* player = PLAYER;
+
+    this->struct_300.unk_18 = player->actor.posRot.pos;
+    this->struct_300.unk_14 = kREG(16) + 9.0f;
+    func_80034A14(&this->actor, &this->struct_300, kREG(17) + 0xC, 4);
+}
+
+void func_80AB10C4(EnNb* this) {
+    s32 pad2[2];
+    Vec3s* tempPtr;
+    Vec3s* tempPtr2;
+
+    tempPtr = &this->struct_300.unk_08;
+    Math_SmoothScaleMaxMinS(&tempPtr->x, 0, 20, 6200, 100);
+    Math_SmoothScaleMaxMinS(&tempPtr->y, 0, 20, 6200, 100);
+    tempPtr2 = &this->struct_300.unk_0E;
+    Math_SmoothScaleMaxMinS(&tempPtr2->x, 0, 20, 6200, 100);
+    Math_SmoothScaleMaxMinS(&tempPtr2->y, 0, 20, 6200, 100);
+}
+
+void EnNb_UpdateEyes(EnNb* this) {
+    s32 pad[3];
+    s16* blinkTimer = &this->blinkTimer;
+    s16* eyeIdx = &this->eyeIdx;
+
+    if (!DECR(*blinkTimer)) {
+        *blinkTimer = Math_Rand_S16Offset(60, 60);
+    }
+
+    *eyeIdx = *blinkTimer;
+    if (*eyeIdx >= ARRAY_COUNT(sEyeSegments)) {
+        *eyeIdx = 0;
+    }
+}
+
+void func_80AB11EC(EnNb* this) {
+    this->action = NB_ACTION_7;
+    this->drawMode = NB_DRAW_NOTHING;
+    this->alpha = 0;
+    this->flag = 0;
+    this->actor.shape.unk_14 = 0;
+    this->alphaTimer = 0.0f;
+}
+
+void func_80AB1210(EnNb* this, GlobalContext* globalCtx) {
+    s32 one; // required to match
+    if (globalCtx->csCtx.state == 0) {
+        if (D_80AB4318) {
+            if (this->actor.params == NB_TYPE_DEMO02) {
+                func_80AB11EC(this);
+            }
+
+            D_80AB4318 = 0;
+        }
+    } else {
+        one = 1;
+        if (!D_80AB4318) {
+            D_80AB4318 = one;
+        }
+    }
+}
+
+void func_80AB1284(EnNb* this, GlobalContext* globalCtx) {
+    func_8002E4B4(globalCtx, &this->actor, 75.0f, 30.0f, 30.0f, 4);
+}
+
+s32 EnNb_FrameUpdateMatrix(EnNb* this) {
+    return SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+CsCmdActorAction* EnNb_GetNpcCsAction(GlobalContext* globalCtx, s32 npcActionIdx) {
+    if (globalCtx->csCtx.state != 0) {
+        return globalCtx->csCtx.npcActions[npcActionIdx];
+    }
+    return NULL;
+}
+
+void EnNb_SetupCsPosRot(EnNb* this, GlobalContext* globalCtx, s32 npcActionIdx) {
+    CsCmdActorAction* csCmdNPCAction = EnNb_GetNpcCsAction(globalCtx, npcActionIdx);
+    s16 newRotY;
+    Actor* thisx = &this->actor;
+
+    if (csCmdNPCAction != NULL) {
+        thisx->posRot.pos.x = csCmdNPCAction->startPos.x;
+        thisx->posRot.pos.y = csCmdNPCAction->startPos.y;
+        thisx->posRot.pos.z = csCmdNPCAction->startPos.z;
+        newRotY = csCmdNPCAction->rot.y;
+        thisx->shape.rot.y = newRotY;
+        thisx->posRot.rot.y = newRotY;
+    }
+}
+
+s32 func_80AB1390(EnNb* this, GlobalContext* globalCtx, u16 arg2, s32 npcActionIdx) {
+    CsCmdActorAction* csCmdNPCAction;
+    if ((globalCtx->csCtx.state != 0) &&
+        (csCmdNPCAction = globalCtx->csCtx.npcActions[npcActionIdx], csCmdNPCAction != NULL) &&
+        (csCmdNPCAction->action == arg2)) {
+        return 1;
+    }
+    return 0;
```

Comment:
```suggestion
        return true;
    }
    return false;
```
and for some other functions too